packages/frontend/mobile/src/views/all-docs/doc/masonry.tsx

```diff
@@ -0,0 +1,65 @@
+import type { DocMeta } from '@blocksuite/store';
+import { useCallback, useEffect, useRef, useState } from 'react';
+
+import { DocCard } from '../../../components';
+import * as styles from './masonry.css';
+
+// TODO(@CatsJuice): Large amount docs performance
```

Comment:
may need to introduce a 3rd party library for this one later because it is also planned for desktop as well.

Reply:
I have no idea. Maybe there isn't any that supports virtual rendering. However lazy render + auto layout should be widely available.